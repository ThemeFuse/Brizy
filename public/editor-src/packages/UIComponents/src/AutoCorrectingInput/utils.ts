import { clamp } from "../utils/math";

export function formatInputValue(value: number, step: number): string | number {
  const stepDecimalLength = decimalLength(step);

  return stepDecimalLength > 0
    ? value.toFixed(stepDecimalLength)
    : String(value);
}

export function decimalLength(number: number): number {
  const [, decimal] = String(number).split(".");
  return decimal ? decimal.length : 0;
}

export function correctNumber(
  number: number,
  min: number,
  max: number,
  step: number
): number {
  return roundByStep(clamp(number, min, max), step, min);
}

export const preciseAdd = preciseOperation.bind(null, "add");

export const preciseSub = preciseOperation.bind(null, "sub");

function roundByStep(number: number, step: number, offset = 0) {
  // this is done to avoid quirks with floating point operations
  // like 1.1 - 1 = 0.10000000000000009
  const mul = 10 ** decimalLength(step);

  number = number * mul;
  step = step * mul;
  offset = offset * mul;

  const tmp = Math.ceil((number - offset) / step) * step + offset;

  return tmp / mul;
}

function preciseOperation(type: "add" | "sub", a: number, b: number) {
  const dla = decimalLength(a);
  const dlb = decimalLength(b);
  const dlMax = Math.max(dla, dlb);
  const mul = 10 ** dlMax;

  const a_ = a * mul;
  const b_ = b * mul;
  let res;

  switch (type) {
    case "add":
      res = a_ + b_;
      break;
    case "sub":
      res = a_ - b_;
      break;
    default:
      throw new Error(`operation type unknown ${type}`);
  }

  return res / mul;
}
