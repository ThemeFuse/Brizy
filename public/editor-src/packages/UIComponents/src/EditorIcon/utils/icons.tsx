import {
  CheckCircle,
  CheckSmall,
  Circle2,
  CircleRemove,
  Close,
  Duplicate,
  Heart,
  MediaImage,
  MediaMap,
  MediaVideo,
  MinistryBrandsEventCalendar,
  MinistryBrandsEventDetail,
  MinistryBrandsEventFeatured,
  MinistryBrandsEventLayout,
  MinistryBrandsEventList,
  MinistryBrandsGroupDetail,
  MinistryBrandsGroupFeatured,
  MinistryBrandsGroupLayout,
  MinistryBrandsGroupList,
  MinistryBrandsGroupSlider,
  MinistryBrandsSermonDetail,
  MinistryBrandsSermonFeatured,
  MinistryBrandsSermonLayout,
  MinistryBrandsSermonList,
  PlusCircle,
  Reverse,
  Settings,
  Spinner,
  StreDown,
  StreUp,
  TextAlignCenter,
  TextAlignLeft,
  TextAlignRight
} from "../../../../icons/src";
import { IconsName } from "../types";

export const Icons = {
  [IconsName.CheckSmall]: CheckSmall,
  [IconsName.MediaImage]: MediaImage,
  [IconsName.MediaVideo]: MediaVideo,
  [IconsName.MediaMap]: MediaMap,
  [IconsName.Heart]: Heart,
  [IconsName.Spinner]: Spinner,
  [IconsName.TextAlignCenter]: TextAlignCenter,
  [IconsName.TextAlignLeft]: TextAlignLeft,
  [IconsName.TextAlignRight]: TextAlignRight,
  [IconsName.PlusCircle]: PlusCircle,
  [IconsName.Reverse]: Reverse,
  [IconsName.Duplicate]: Duplicate,
  [IconsName.CheckCircle]: CheckCircle,
  [IconsName.Settings]: Settings,
  [IconsName.Close]: Close,
  [IconsName.StreUp]: StreUp,
  [IconsName.StreDown]: StreDown,
  [IconsName.Circle2]: Circle2,
  [IconsName.CircleRemove]: CircleRemove,
  [IconsName["t2-sermon-list"]]: MinistryBrandsSermonList,
  [IconsName["t2-sermon-detail"]]: MinistryBrandsSermonDetail,
  [IconsName["t2-sermon-featured"]]: MinistryBrandsSermonFeatured,
  [IconsName["t2-sermon-layout"]]: MinistryBrandsSermonLayout,
  [IconsName["t2-event-list"]]: MinistryBrandsEventList,
  [IconsName["t2-event-detail"]]: MinistryBrandsEventDetail,
  [IconsName["t2-event-featured"]]: MinistryBrandsEventFeatured,
  [IconsName["t2-event-layout"]]: MinistryBrandsEventLayout,
  [IconsName["t2-event-calendar"]]: MinistryBrandsEventCalendar,
  [IconsName["t2-group-list"]]: MinistryBrandsGroupList,
  [IconsName["t2-group-detail"]]: MinistryBrandsGroupDetail,
  [IconsName["t2-group-featured"]]: MinistryBrandsGroupFeatured,
  [IconsName["t2-group-slider"]]: MinistryBrandsGroupSlider,
  [IconsName["t2-group-layout"]]: MinistryBrandsGroupLayout,
  [IconsName["nc-stre-up"]]: StreUp,
  [IconsName["nc-stre-down"]]: StreDown
};
