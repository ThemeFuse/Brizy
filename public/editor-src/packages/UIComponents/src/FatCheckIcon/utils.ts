export const getActiveClassName = (
  className: string,
  isActive?: boolean
): string => `${className}${isActive ? "active" : "inactive"}`;

export const getCheckedClassName = (
  className: string,
  isChecked?: boolean
): string => `${className}${isChecked ? "checked" : "unchecked"}`;
