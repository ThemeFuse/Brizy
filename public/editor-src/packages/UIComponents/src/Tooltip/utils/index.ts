type PositionState = "above" | "below";

const positionState: PositionState = "above";

export const getPosition = (): PositionState => positionState;
