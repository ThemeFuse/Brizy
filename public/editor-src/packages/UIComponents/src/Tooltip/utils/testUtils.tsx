import { ToolbarItemsInstance } from "../types";
import { Placement, Size } from "../types";

export const getToolbar = () =>
  ({
    toolbarRef: { current: document.createElement("div") },
    toolbarCSSPosition: "absolute",
    toolbarItemIndex: 0,
    toolbarItemsLength: 1
  } as unknown as ToolbarItemsInstance);

export const sizes: Size[] = ["auto", "small", "medium", "large", "xlarge"];
export const placements: Placement[] = [
  "auto",
  "auto-start",
  "auto-end",
  "top",
  "top-start",
  "top-end",
  "bottom",
  "bottom-start",
  "bottom-end",
  "right",
  "right-start",
  "right-end",
  "left",
  "left-start",
  "left-end"
];
export const getOffsetPosition = (placement: Omit<Placement, "auto">) => {
  switch (placement) {
    case "right-start":
    case "right-end":
    case "right":
      return "left";
    case "left-start":
    case "left-end":
    case "left":
      return "right";
    case "top-start":
    case "top-end":
    case "top":
      return "bottom";
    case "bottom-start":
    case "bottom-end":
    case "bottom":
      return "top";
  }
};
