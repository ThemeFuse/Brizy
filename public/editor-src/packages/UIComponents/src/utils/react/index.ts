import { SyntheticEvent } from "react";

type WithValue = HTMLElement & { value: string };

export const inputValue = (e: SyntheticEvent<WithValue>): string =>
  (e.target as WithValue).value;
