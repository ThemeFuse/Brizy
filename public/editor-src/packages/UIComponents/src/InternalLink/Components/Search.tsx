import classNames from "classnames";
import React from "react";
import { EditorIcon } from "../../EditorIcon";
import { IconsName } from "../../EditorIcon/types";
import { t } from "../../utils/i18n";
import { SearchProps } from "./types";

export const Search: React.FC<SearchProps> = ({ isLoading, onChange }) => {
  const spinnerClassName = classNames("brz-ed-control__internalLink__spinner", {
    "brz-ed-control__internalLink__spinner--hidden": !isLoading,
    "brz-ed-animated--spin": isLoading
  });

  return (
    <div className="brz-ed-control__internalLink__search">
      <input
        className="brz-input"
        placeholder={t("Type to Search ...")}
        autoFocus
        spellCheck={false}
        onChange={onChange && ((e): void => onChange(e.target.value))}
      />
      <EditorIcon icon={IconsName.Circle2} className={spinnerClassName} />
    </div>
  );
};
