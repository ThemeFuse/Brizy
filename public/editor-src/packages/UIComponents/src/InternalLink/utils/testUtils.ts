import { waitFor } from "@testing-library/react";
import { Choice } from "../types";

export const waitIcon = async (container: HTMLElement) =>
  await waitFor(() =>
    expect(container.querySelector(".brz-icon-svg")).toBeInTheDocument()
  );

export const items: Choice[] = [
  { title: "first", value: 1 },
  { title: "second", value: 2 },
  { title: "third", value: 3 },
  { title: "fourth", value: 4 },
  { title: "fifth", value: 5 },
  { title: "sixth", value: 6 },
  { title: "seventh", value: 7 },
  { title: "eighth", value: 8 },
  { title: "ninth", value: 9 },
  { title: "tenth", value: 10 },
  { title: "eleventh", value: 11 },
  { title: "twelfth", value: 12 },
  { title: "thirteenth", value: 13 },
  { title: "fourteenth", value: 14 },
  { title: "fifteenth", value: 15 }
];
