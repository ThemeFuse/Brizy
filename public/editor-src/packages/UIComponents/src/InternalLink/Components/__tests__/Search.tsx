import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { waitIcon } from "../../utils/testUtils";
import { Search } from "../Search";

describe("Test Search", () => {
  describe("Snapshots and Props", () => {
    test("loading shouldn't be", async () => {
      const { container } = render(<Search />);

      await waitIcon(container);

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(
          ".brz-ed-control__internalLink__spinner--hidden"
        )
      ).toBeInTheDocument();
      expect(
        container.querySelector(".brz-ed-animated--spin")
      ).not.toBeInTheDocument();
    });

    test("isLoading should be true", async () => {
      const { container } = render(<Search isLoading />);

      expect(container.firstChild).toMatchSnapshot();

      await waitIcon(container);

      expect(
        container.querySelector(
          ".brz-ed-control__internalLink__spinner--hidden"
        )
      ).not.toBeInTheDocument();
      expect(
        container.querySelector(".brz-ed-animated--spin")
      ).toBeInTheDocument();
    });
  });

  describe("Event Handlers", () => {
    test("Check onChange event handler", () => {
      const onChange = jest.fn();
      const { container } = render(<Search onChange={onChange} />);
      const input = container.querySelector(".brz-input");

      input && fireEvent.change(input, { target: { value: "value" } });

      expect(onChange).toHaveBeenCalledWith("value");
    });
  });
});
