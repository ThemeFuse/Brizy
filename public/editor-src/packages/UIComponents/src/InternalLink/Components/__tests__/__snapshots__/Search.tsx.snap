// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search Snapshots and Props isLoading should be true 1`] = `
<div
  class="brz-ed-control__internalLink__search"
>
  <input
    class="brz-input"
    placeholder="Type to Search ..."
    spellcheck="false"
  />
</div>
`;

exports[`Test Search Snapshots and Props loading shouldn't be 1`] = `
<div
  class="brz-ed-control__internalLink__search"
>
  <input
    class="brz-input"
    placeholder="Type to Search ..."
    spellcheck="false"
  />
  <svg
    class="brz-icon-svg brz-ed-icon-svg align-[initial] brz-ed-control__internalLink__spinner brz-ed-control__internalLink__spinner--hidden"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        class="nc-loop_circle-02-16"
        transform="rotate(359.71071426823204 8 8)"
      >
        <path
          d="M8,16c-4.4111328,0-8-3.5888672-8-8s3.5888672-8,8-8s8,3.5888672,8,8S12.4111328,16,8,16z M8,2C4.6914062,2,2,4.6914062,2,8s2.6914062,6,6,6s6-2.6914062,6-6S11.3085938,2,8,2z"
          fill="currentColor"
          opacity="0.4"
        />
        <path
          d="M16,8h-2c0-3.3085938-2.6914062-6-6-6V0C12.4111328,0,16,3.5888672,16,8z"
          data-color="color-2"
        />
      </g>
      <script>
        !function(){function t(t){this.element=t,this.animationId,this.start=null,this.init()}if(!window.requestAnimationFrame){var i=null;window.requestAnimationFrame=function(t,n){var e=(new Date).getTime();i||(i=e);var a=Math.max(0,16-(e-i)),o=window.setTimeout(function(){t(e+a)},a);return i=e+a,o}}t.prototype.init=function(){var t=this;this.animationId=window.requestAnimationFrame(t.triggerAnimation.bind(t))},t.prototype.reset=function(){var t=this;window.cancelAnimationFrame(t.animationId)},t.prototype.triggerAnimation=function(t){var i=this;this.start||(this.start=t);var n=t-this.start;504&gt;n||(this.start=this.start+504),this.element.setAttribute("transform","rotate("+Math.min(n/1.4,360)+" 8 8)");if(document.documentElement.contains(this.element))window.requestAnimationFrame(i.triggerAnimation.bind(i))};var n=document.getElementsByClassName("nc-loop_circle-02-16"),e=[];if(n)for(var a=0;n.length&gt;a;a++)!function(i){e.push(new t(n[i]))}(a);document.addEventListener("visibilitychange",function(){"hidden"==document.visibilityState?e.forEach(function(t){t.reset()}):e.forEach(function(t){t.init()})})}();
      </script>
    </g>
  </svg>
</div>
`;
